flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MySqlOnLineSchemaMigrationITCase.java

```diff
@@ -0,0 +1,591 @@
+                    : "https://github.com/github/gh-ost/releases/download/v1.1.6/gh-ost-binary-linux-arm64-20231207144046.tar.gz";
+
+    @BeforeClass
+    public static void beforeClass() {
+        LOG.info("Starting MySql8 containers...");
+        Startables.deepStart(Stream.of(MYSQL8_CONTAINER)).join();
+        Startables.deepStart(Stream.of(PERCONA_TOOLKIT_CONTAINER)).join();
+        LOG.info("Container MySql8 is started.");
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        LOG.info("Stopping MySql8 containers...");
+        MYSQL8_CONTAINER.stop();
+        PERCONA_TOOLKIT_CONTAINER.stop();
+        LOG.info("Container MySql8 is stopped.");
+    }
+
+    @Before
+    public void before() {
+        customerDatabase.createAndInitialize();
+        TestValuesTableFactory.clearAllData();
+        env.setParallelism(4);
+        env.enableCheckpointing(200);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+    }
+
+    @After
+    public void after() {
+        customerDatabase.dropDatabase();
+    }
+
+    private static void installGhOstCli(Container<?> container) {
+        try {
+            execInContainer(
+                    container,
+                    "download gh-ost tarball",
+                    "curl",
+                    "-L",
+                    "-o",
+                    "/tmp/gh-ost.tar.gz",
+                    GH_OST_DOWNLOAD_LINK);
+            execInContainer(
+                    container, "unzip binary", "tar", "-xzvf", "/tmp/gh-ost.tar.gz", "-C", "/bin");
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static GenericContainer<?> createPerconaToolkitContainer() {
+        GenericContainer<?> perconaToolkit =
+                new GenericContainer<>(PERCONA_TOOLKIT)
+                        // keep container alive
+                        .withCommand("tail", "-f", "/dev/null")
+                        .withNetwork(NETWORK)
+                        .withLogConsumer(new Slf4jLogConsumer(LOG));
+        return perconaToolkit;
+    }
+
+    @Test
+    public void testGhOstSchemaMigrationFromScratch() throws Exception {
+        LOG.info("Step 1: Install gh-ost command line utility");
+        installGhOstCli(MYSQL8_CONTAINER);
+
+        LOG.info("Step 2: Start pipeline job");
+        env.setParallelism(1);
+        TableId tableId = TableId.tableId(customerDatabase.getDatabaseName(), "customers");
+        MySqlSourceConfigFactory configFactory =
+                new MySqlSourceConfigFactory()
+                        .hostname(MYSQL8_CONTAINER.getHost())
+                        .port(MYSQL8_CONTAINER.getDatabasePort())
+                        .username(TEST_USER)
+                        .password(TEST_PASSWORD)
+                        .databaseList(customerDatabase.getDatabaseName())
+                        .tableList(customerDatabase.getDatabaseName() + "\\.customers")
```

Comment:
I think it would be more reliable to validate capturing the changes of entire database and making twice online table change changes.